src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -393,6 +408,19 @@
       (catch CRLException e
         e))))
 
+(schema/defn set-ca-config-defaults
+  "Sets defaults for any of the missing config parameters for infra CRL"
+  [infra-crl-config
+   cadir]
+  ;; TODO - We want to use the current node/certificate as the default in here instead of starting empty
+  ;; Open - Is there a way to eliminate these temps below? Cannot quite use an ->
+  (let [t1 (if (nil? (:compile-masters infra-crl-config)) (merge infra-crl-config {:compile-masters []}) infra-crl-config)
```

Comment:
When I mentioned using merge before I was thinking something more along the lines of (I think you were using the threading macro before, but creating an intermediate variable would work too):
```clj
(let [defaults {:compile-masters []
                :compile-masters-serials-path (str cadir "/cm-serials.txt")
                :infra-crl-path (str cadir "/infra-crl.pem")
                :enable-infra-crl true}]
  (-> infra-crl-config
    (merge defaults)
    (dissoc :certificate-status)))
```

Reply:
Thanks! I knew there had to be a better way. Though the order of args to merge matters. I changed it to have the intended behavior of explicit values over riding defaults.